packages/nuxt/test/runtime/plugins/server.test.ts

```diff
@@ -1,64 +1,33 @@
-import { afterEach, describe, expect, it, vi } from 'vitest';
+import { getTraceMetaTags } from '@sentry/core';
+import { type Mock, afterEach, describe, expect, it, vi } from 'vitest';
 import { addSentryTracingMetaTags } from '../../../src/runtime/utils';
 
-const mockReturns = vi.hoisted(() => {
-  return {
-    traceHeader: 'trace-header',
-    baggageHeader: 'baggage-header',
-  };
-});
-
-vi.mock('@sentry/core', async () => {
-  const actual = await vi.importActual('@sentry/core');
-
-  return {
-    ...actual,
-    getActiveSpan: vi.fn().mockReturnValue({ spanId: '123' }),
-    getRootSpan: vi.fn().mockReturnValue({ spanId: 'root123' }),
-    spanToTraceHeader: vi.fn(() => mockReturns.traceHeader),
-  };
-});
-
-vi.mock('@sentry/opentelemetry', async () => {
-  const actual = await vi.importActual('@sentry/opentelemetry');
-
-  return {
-    ...actual,
-    getDynamicSamplingContextFromSpan: vi.fn().mockReturnValue('contextValue'),
-  };
-});
-
-vi.mock('@sentry/utils', async () => {
-  const actual = await vi.importActual('@sentry/utils');
-
-  return {
-    ...actual,
-    dynamicSamplingContextToSentryBaggageHeader: vi.fn().mockReturnValue(mockReturns.baggageHeader),
-  };
-});
+vi.mock('@sentry/core', () => ({
+  getTraceMetaTags: vi.fn(),
+}));
 
 describe('addSentryTracingMetaTags', () => {
   afterEach(() => {
     vi.resetAllMocks();
   });
 
-  it('should add meta tags when there is an active root span', () => {
+  it('should add meta tags to the head array', () => {
+    const mockMetaTags = [
+      '<meta name="sentry-trace" content="12345678901234567890123456789012-1234567890123456-1"/>',
+      '<meta name="baggage" content="sentry-environment=production"/>',
+    ].join('\n');
+
+    // return value is mocked here as return values of `getTraceMetaTags` are tested separately (in @sentry/core)
+    (getTraceMetaTags as Mock).mockReturnValue(mockMetaTags);
```

Comment:
I think that's fine, given the unit under test here is the Nuxt-specific `addSentryTracingMetaTags` function. 

What we should do in the near future is to make sure our e2e tests cover distributed tracing behaviour. For example, we should check that 
- the sever-side request transaction and the client-side pageload transaction is connected by the same trace. This will validate that the trace meta tag injection generally works.
- potential requests made on the client side to the server (thinking of API endpoint or data loading calls, whatever nuxt offers) are connected to the server side. This is probably easiest testable when making these calls during or directly after a navigation so that we can make sure a client-side root span exists.

I'm also working on a SvelteKit e2e test app that's specifically configured for tracing without performance. We could think about doing something similar for Nuxt but I wouldn't require it for the moment.